services/historic_prices.go

```diff
@@ -17,6 +17,53 @@ func StartHistoricPriceService() {
 		time.Sleep(time.Hour)
 	}
 }
+
+func WriteHistoricPricesForDay(day int64) error {
```

Comment:
Why isn't this function also used here: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/2880fa2d8ce2fb31c1dadd7445eb457f96272d01/services/historic_prices.go#L90-L118

The only difference seems to be the addition of the `ON CONFLICT` clause. This could be a parameter though. That way you would reduce duplicated code.

What are your thoughts on that?

Reply:
The `ON CONFLICT` clause doesn't really matter, the `updateHistoricPrices` could also use it.
The issue is the update of `currentDay` which is different but it can actually be refactored.